src/dlgTriggerEditor.cpp

```diff
@@ -5520,19 +5537,51 @@ void dlgTriggerEditor::setupPatternControls(const int type, dlgTriggerPatternEdi
 {
     // Display middle dots for potentially unwanted spaces in perl regex
     if (type == REGEX_PERL) {
-        markQLineEdit(pItem->lineEdit_pattern);
+        markQTextEdit(pItem->lineEdit_pattern);
         lineEditShouldMarkSpaces[pItem->lineEdit_pattern] = true;
+        pItem->lineEdit_pattern->blockSignals(true);
+        pItem->lineEdit_pattern->rehighlight();
+        pItem->lineEdit_pattern->blockSignals(false);
     } else {
-        unmarkQLineEdit(pItem->lineEdit_pattern);
+        unmarkQTextEdit(pItem->lineEdit_pattern);
         lineEditShouldMarkSpaces[pItem->lineEdit_pattern] = false;
     }
 
     switch (type) {
     case REGEX_SUBSTRING:
+        pItem->lineEdit_pattern->setHighlightingEnabled(false);
+        pItem->lineEdit_pattern->show();
+        pItem->pushButton_fgColor->hide();
+        pItem->pushButton_bgColor->hide();
+        pItem->label_prompt->hide();
+        pItem->spinBox_lineSpacer->hide();
+        break;
     case REGEX_PERL:
+        pItem->lineEdit_pattern->setHighlightingEnabled(true);
+        pItem->lineEdit_pattern->show();
+        pItem->pushButton_fgColor->hide();
+        pItem->pushButton_bgColor->hide();
+        pItem->label_prompt->hide();
+        pItem->spinBox_lineSpacer->hide();
+        break;
     case REGEX_BEGIN_OF_LINE_SUBSTRING:
+        pItem->lineEdit_pattern->setHighlightingEnabled(false);
+        pItem->lineEdit_pattern->show();
+        pItem->pushButton_fgColor->hide();
+        pItem->pushButton_bgColor->hide();
+        pItem->label_prompt->hide();
+        pItem->spinBox_lineSpacer->hide();
+        break;
     case REGEX_EXACT_MATCH:
+        pItem->lineEdit_pattern->setHighlightingEnabled(false);
+        pItem->lineEdit_pattern->show();
+        pItem->pushButton_fgColor->hide();
+        pItem->pushButton_bgColor->hide();
+        pItem->label_prompt->hide();
+        pItem->spinBox_lineSpacer->hide();
+        break;
     case REGEX_LUA_CODE:
+        pItem->lineEdit_pattern->setHighlightingEnabled(false);
```

Comment:
The change can be made a bit easier to track by doing it like this - though we are missing the  desirable `[[fallthough]];` between each, except the last, `case REGEX_XXXX`:
```suggestion
    case REGEX_SUBSTRING:
    case REGEX_PERL:
    case REGEX_BEGIN_OF_LINE_SUBSTRING:
    case REGEX_EXACT_MATCH:
    case REGEX_LUA_CODE:
        pItem->lineEdit_pattern->setHighlightingEnabled(type == REGEX_PERL);
```

Reply:
very nice, done